pandas/_libs/interval.pyx

```diff
@@ -229,7 +231,7 @@ def _warning_interval(inclusive: str | None = None, closed: None | lib.NoDefault
             stacklevel=2,
         )
         if closed is None:
-            inclusive = "both"
+            inclusive = "right"
```

Comment:
not sure about some of the handling/validation here but that's outside the scope of this PR.

we have a `deprecate_kwarg` decorator that caters for renaming keywords. not sure why not used.

Reply:
I forgot about it, not sure why it was not used before. Switched over now.